linux/src/main/cpp/DarkModeGnome.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Jannis Weis
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "com_github_weisj_darkmode_platform_linux_gnome_GnomeNative.h"
+
+#include <string>
+#include <glibmm-2.4/glibmm.h>
+#include <giomm-2.4/giomm.h>
+
+JNIEXPORT jstring JNICALL
+Java_com_github_weisj_darkmode_platform_linux_gnome_GnomeNative_getCurrentTheme(JNIEnv *env, jclass) {
+  Gio::init();
```

Comment:
The [documentation](https://developer.gnome.org/glibmm/stable/classGio_1_1Initable.html#a5881cf9c259d5af218585b7ecde38464) for `Gio::init` states that

> Callers should not assume that a class which implements Initable can be initialized multiple times

I propose to move the calls to `Gio::init()` to an separate method, which gets called by `GnomeThemeMonitorService#install`. 

Also `Glib::RefPtr<Gio::Settings> settings` could probably go into the global scope (with initialization in the same method as  `Gio::init()` gets called) as we only need one instance anyway (and there is no harm done in using global variables in a single file library). This way the seetings wouldn't need to be obtained each time `getCurrentTheme()` is called and no reference needs to be stored in `EventHandler`.

Reply:
I've done this in b28c5aaba365b270d7c2d85c9ece94fc261c7fde. However, this results in a crash on IntelliJ startup as there was a settings call before `Gio::init()` is called. Despite the install method being available, `AutoDarkModeStartupListener.appFrameCreated()` does not call `install()` on the service it retrieves, causing `isDarkThemeEnabled()` to be called before `install()`. It seems that only `AutoDarkModePluginListener.pluginLoaded()` calls `install()`. Should this be changed? I'm unfamiliar with the IDEA plugin lifecycle.

Reply:
I've done this in b28c5aaba365b270d7c2d85c9ece94fc261c7fde. However, this results in a crash on IntelliJ startup as there was a settings call before `Gio::init()` is called. Despite the install method being available, `AutoDarkModeStartupListener.appFrameCreated()` does not call `install()` on the service it retrieves, causing `isDarkThemeEnabled()` to be called before `install()`. It seems that only `AutoDarkModePluginListener.pluginLoaded()` calls `install()`. Should this be changed? I'm unfamiliar with the IDEA plugin lifecycle.

Reply:
Yes I have forgotten my initial intend for the `#install` method. The method was added because the macOS implementation needed to do a one time job after the plugin has been installed (Some hacky operations on the applications bundle). Though there should be no harm in calling `#install` each time on startup. I would propose the following changes:
- Rename `AutoDarkMode#install` and AutoDarkMode#uninstall` to `AutoDarkMode#pluginLoaded` and   AutoDarkMode#pluginUnloaded` and remove the calls to `ThemeMonitor#install/uninstall`.
- Remove `ThemeMonitor#uninstall` as there is no need for it there is.
- Call `ThemeMonitor#install` after `AbstractThemeMonitor` is created.

Reply:
Changed with 60038760189573637dc234e7459ee9fced66384b.